elf_binary.cc

```diff
@@ -156,7 +156,7 @@ const Elf_Phdr& ELFBinary::GetPhdr(uint64_t type) {
     return *phdr;
 }
 
-// GetVerneed returns (soname, version)
+// GetVerneed returns (filename, version)
```

Comment:
Now it makes no sense to return `filename` as it can be obtained by `ELFBinary::name()`? I'm not sure, but if we cannot use the information in verdef, I guess we should correct the caller?

Reply:
The problem is we have returned a filename when GetVersion function finds the corresponding filed in Elf_Verneed. This is because we cannot get the soname from the Elf_Verneed or Elf_Vernaux.

So, I decided to split this PR into two PR. The first one is [PR36](https://github.com/shinh/sold/pull/36/files). It fixes the DT_NEEDED fields. The second one, which I am making now, fixes GetVersion function.

Reply:
This is the second one. https://github.com/shinh/sold/pull/37